### DISCUSSION TITLE
Return stale values if there's an exception thrown

### DISCUSSION THREAD

**Discussion by jonaspero** [ben-manes/caffeine#1800] [Q&A] [answered]
Hi,

I use Caffeine combined with Spring Boot.
My current code looks like this:

```
    @Bean("tenMinuteCache")
    fun shortCache(): CacheManager =
        object : ConcurrentMapCacheManager() {
            override fun createConcurrentMapCache(name: String): Cache {
                val concurrentMap =
                    Caffeine
                        .newBuilder()
                        .maximumSize(1000)
                        .expireAfterWrite(10, TimeUnit.MINUTES)
                        .recordStats()
                        .build<Any, Any>()
                        .asMap()
                return ConcurrentMapCache(name, concurrentMap, true)
            }
        }


    @Cacheable("alle-country", cacheManager = "tenMinuteCache")
    fun retrieveCountry(): CountryDto {
        val uri = URI.create("$uri/countries")

        return getCountry {
            getForEntity(uri)
        }
    }
```

The issue is that sometimes retrieveCountry will throw an exception.
Is there a way to return stale, last value from the cache if this happens?

**Comment by ben-manes** [comment, marked as answer]
It sounds like you want to also use [refreshAfterWrite](https://github.com/ben-manes/caffeine/wiki/Refresh). This allows the entry to be reloaded asynchronously and return the stale value, while using expiration to evict if inactive and beyond a usability threshold. Unfortunately this is not supported by Spring (https://github.com/spring-projects/spring-boot/issues/41762), so you would need to use our APIs directly.
